### PR TITLE
Kdo push command

### PR DESCRIPTION
**Pull request by johnmcollier**
## What is the purpose of this change? What does it change?

- Updates the `kdo push` command to create the PersistentVolumeClaim when creating the runtime container and sync the source code to it
    - Mounts the volume under `/projects`
    - Attaches the volume to the runtime container
    - Syncs the source code to the runtime container. A separate PR will sync to a dedicated sync container that mounts the volume
- Implements unit tests for the following packages: `storage`, `labels`, and `volumes`
- Some cleanup:
    - Changes instances of `DeploymentConfig` to `Deployment`
    - Changes instances of `dc` to `dep`

## Was the change discussed in an issue?
fixes #???
N/A

### CODE REVIEW

**Comment by johnmcollier** [comment]
```
johns-mbp-3:nodejs-ex johncollier$ kdo push
Validation
 ✓  Validating component [400ms]
 ✓  Checking component [37ms]

Configuration changes
 ✓  Added storage nodejs-idp-nodejs-ex-pgko to nodejs-idp-nodejs-ex-pgko
 ✓  Initializing component
 ✓  Creating component [185ms]

Pushing to component nodejs-idp-nodejs-ex-pgko of type local
 ✓  Waiting for component to start [3s]
 ✓  Copying files to component [3s]
 ✓  Changes successfully pushed to component
```

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+
+}
+
+// Create adds storage to given component of given application
+func Create(client *kclient.Client, name string, size string, componentName string, applicationName string) (*corev1.PersistentVolumeClaim, error) {
+
+	// Namespace the component
+	// We will use name+applicationName instead of componentName+applicationName until:
+	// https://github.com/openshift/odo/issues/504 is resolved.
+	namespacedOpenShiftObject, err := util.NamespaceKubernetesObject(name, applicationName)
```
Comment (rajivnathan) [inline pkg/storage/storage.go]: nit: `namespacedOpenShiftObject` -> `namespacedKubernetesObject`

File: pkg/storage/storage.go [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	labels := storagelabels.GetLabels(name, componentName, applicationName, true)
+
+	glog.V(4).Infof("Got labels for PVC: %v", labels)
+
+	// Create PVC
+	pvc, err := client.CreatePVC(generatePVCNameFromStorageName(namespacedOpenShiftObject), size, labels)
```
Comment (rajivnathan) [inline pkg/storage/storage.go]: nit: `namespacedOpenShiftObject` -> `namespacedKubernetesObject`

File: pkg/storage/storage.go [hunk 3]
```diff
@@ -0,0 +1,460 @@
...
+func Mount(client *kclient.Client, path string, storageName string, componentName string, applicationName string) error {
+	storageComponent, err := GetComponentNameFromStorageName(client, storageName)
+	if err != nil {
+		return errors.Wrap(err, "unable to get the component name associated with the storage")
+	}
+	if storageComponent != "" {
+		return fmt.Errorf("the given storage is already mounted to the component %v", storageComponent)
+	}
+
+	namespacedOpenShiftObject, err := util.NamespaceKubernetesObject(storageName, applicationName)
```
Comment (rajivnathan) [inline pkg/storage/storage.go]: nit: `namespacedOpenShiftObject` -> `namespacedKubernetesObject`

File: pkg/storage/storage.go [hunk 4]
```diff
@@ -0,0 +1,460 @@
...
+	if err != nil {
+		return errors.Wrapf(err, "unable to create namespaced name")
+	}
+
+	pvc, err := client.GetPVCFromName(generatePVCNameFromStorageName(namespacedOpenShiftObject))
```
Comment (rajivnathan) [inline pkg/storage/storage.go]: nit: `namespacedOpenShiftObject` -> `namespacedKubernetesObject`

File: pkg/storage/storage_test.go [hunk 5]
```diff
@@ -0,0 +1,847 @@
...
+		{
+			name: "test case 2: with multiple pvcs",
+			inputStorage: []Storage{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-pvc-0",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
+						APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (rajivnathan) [inline pkg/storage/storage_test.go]: APIVersion for kdo

File: pkg/storage/storage_test.go [hunk 6]
```diff
@@ -0,0 +1,847 @@
...
+						Path: "data",
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "example-pvc-1",
+					},
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "List",
+						APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (rajivnathan) [inline pkg/storage/storage_test.go]: APIVersion for kdo

File: pkg/storage/storage_test.go [hunk 7]
```diff
@@ -0,0 +1,847 @@
...
+					},
+					Status: StorageStatus{
+						Path: "backend",
+					},
+				},
+			},
+			want: StorageList{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (rajivnathan) [inline pkg/storage/storage_test.go]: APIVersion for kdo

File: pkg/storage/storage_test.go [hunk 8]
```diff
@@ -0,0 +1,847 @@
...
+				},
+				ListMeta: metav1.ListMeta{},
+				Items: []Storage{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "example-pvc-0",
+						},
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "List",
+							APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (rajivnathan) [inline pkg/storage/storage_test.go]: APIVersion for kdo

File: pkg/storage/storage_test.go [hunk 9]
```diff
@@ -0,0 +1,847 @@
...
+							Path: "data",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "example-pvc-1",
+						},
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "List",
+							APIVersion: "odo.openshift.io/v1alpha1",
```
Comment (rajivnathan) [inline pkg/storage/storage_test.go]: APIVersion for kdo

**Pull request merged**